server/internal/web/handler/decoder.go

```diff
@@ -0,0 +1,95 @@
+package handler
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+func DecodeJSONBody(w http.ResponseWriter, r *http.Request, dst interface{}) error {
```

Comment:
This handler (and all other handlers and middlewares) is not tested, I think this is not the right moment to do it due to the complexity of the PR. If we want to merge only tested code I can create a new branch based on the current one and merge this PR only when the next one is ready, otherwise I can open a new issue and we can tackle the problem later.

Reply:
I don't think I would unit-test handlers, we have E2E tests to cover the API as well. Unless we have specific concerns or failure scenarios we want to enforce, Handlers and middlewares are pretty dumb as all the meaty logic is in the domain. Once that is tested, we can reasonably assume they will work as well.